spec/src/main/asciidoc/cdi.asciidoc

```diff
@@ -19,4 +19,158 @@
 
 If a CDI implementation is available, instances of `ManagedExecutor` and `ThreadContext` can be injected as CDI beans.
 
-TODO write the documentation after discussion on Config annotations is resolved
\ No newline at end of file
+=== Injecting `ManagedExecutor`
+
+Application can inject instances of `ManagedExecutor` using CDI with the option to share same executors between multiple injection points.
+The simplest use case will create a new executor instance for every such injection point and the configuration will be equal to default values of `@ManagedExecutorConfig` annotation.
```

Comment:
(This ties into one of the later comments I left about treating the default `@ManagedExecutorConfig` annotation differently than the absence of the annotation entirely, it would be helpful to read that comment first)

If we stick with the pattern that just `@Inject @NamedInstance` declares  _usage_, and `@Inject @NamedInstance @ManagedExecutorConfig` declares _usage and definition_, I think it would make more sense to have 1 shared `ManagedExecutor` across all default injection points (`@Inject ManagedExecutor exec;`).

Since the default ManagedExecutor instance is unbounded (infinite maxAsnyc and infinite maxQueued), it doesn't really matter if we share these instances or not, but conceptually it makes more sense to only get a dedicated instances when you are using `@ManagedExecutorConfig` (which is analogous to a Producer).  

What do others think on this?

Reply:
> Since the default ManagedExecutor instance is unbounded 

That's what we designed now, but note that we discussed https://github.com/eclipse/microprofile-concurrency/issues/31 and external config of default settings. With that in place, defaults could differ (depending on what we allow there, but I see no reason not to permit changing all parameters).

>  I think it would make more sense to have 1 shared ManagedExecutor across all default injection points

Like I said (elsewhere on this PR), this approach is the opposite to what we drafted for sharing. The argument back there was that sharing by default is not intuitive for user. Hence sharing defaults is non-intuitive as well, especially given that in all other cases it behaves differently.

Impl-wise this isn't a problem. You can detect injection points that have a name but no config linked to them. In such case you simply provide default with a name attached to it to enable sharing.